File: backend/app/domain/interview.py
"""Valores cerrados de una entrevista (especificación §4, RF-40…42). Reglas puras."""

from enum import StrEnum


class InterviewType(StrEnum):
    SCREENING = "screening"
    TECHNICAL = "technical"
    HR = "hr"
    CULTURAL = "cultural"
    FINAL = "final"
    OTHER = "other"


class InterviewFormat(StrEnum):
    ONLINE = "online"
    ONSITE = "onsite"
    PHONE = "phone"


class InterviewOutcome(StrEnum):
    """Resultado de la entrevista (RF-41). `PENDING` es el valor por defecto."""

    PENDING = "pending"
    PASSED = "passed"
    FAILED = "failed"
    CANCELLED = "cancelled"

File: frontend/src/features/interviews/types/Interview.ts
export const INTERVIEW_TYPES = [
  "screening",
  "technical",
  "hr",
  "cultural",
  "final",
  "other",
] as const;
export type InterviewType = (typeof INTERVIEW_TYPES)[number];

export const INTERVIEW_FORMATS = ["online", "onsite", "phone"] as const;
export type InterviewFormat = (typeof INTERVIEW_FORMATS)[number];

export const INTERVIEW_OUTCOMES = ["pending", "passed", "failed", "cancelled"] as const;
export type InterviewOutcome = (typeof INTERVIEW_OUTCOMES)[number];

export interface Interview {
  id: string;
  /** Instante ISO con zona. */
  scheduled_at: string;
  duration_minutes: number | null;
  interviewers: string | null;
  interview_type: InterviewType | null;
  format: InterviewFormat | null;
  outcome: InterviewOutcome;
  notes: string | null;
  created_at: string;
  updated_at: string;
}
